### ISSUE TITLE
Investigate Performance

### ISSUE THREAD

**Issue by SAtacker** [STEllAR-GROUP/hpx-rs#13]
HPX is under the process of standardizing it's benchmarks using nano-bench. We need to compare the [overhead introduced by Rust's runtime checks](https://dl.acm.org/doi/fullHtml/10.1145/3551349.3559494) with HPX in cpp.

**Comment by Connor-GH** [comment]
Obviously, the answer will be nonzero. Unless static linking happens, there is no way to optimize across library boundaries.

**Comment by SAtacker** [comment]
> Obviously, the answer will be nonzero. Unless static linking happens, there is no way to optimize across library boundaries.

Yep, all we have to do is document how much of an issue it is in practice.
